src/seedu/addressbook/data/person/Address.java

```diff
@@ -2,17 +2,32 @@
 
 import seedu.addressbook.data.exception.IllegalValueException;
 
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "123, some street, #01-02, 654321";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Addresses are comma-separated fields consisting of the "
+            + "following: BLOCK, STREET, UNIT, POSTAL_CODE\n"
+            + " BLOCK is a number with optional letter suffix,\n"
+            + " STREET is a string not containing commas,\n"
+            + " UNIT is a number,\n"
+            + " POSTAL_CODE is a six-digit number.";
+    public static final String ADDRESS_VALIDATION_REGEX = "^(?<block>[\\d\\w]\\d*\\w*)?(?:, | )?"
```

Comment:
Remember to update the documentation (user guide, help, etc) and tests as well when you modify the behavior.

In particular, you should update the user guide to let the readers know that address need to be in a particular format.